### ISSUE TITLE
Extended support for Spring Controller interfaces

### ISSUE THREAD

**Issue by vghero** [stoicflame/enunciate#1115]
Hey,
I'm using Enunciate 2.11.1 together with a Spring 5 WebMVC REST app. I'm separating the REST controllers via interfaces from its implementation, so that the interface can be reused by client generation like Feign. Recently Feign dropped support for @RequestMapping on Controller interfaces and so I dropped it as well, moving the path to e.g. @GetMapping on method level. Then I also wanted to cleanup @RestController on the interface, as it's already contained on my impl. But then it seemed that enunciate wasn't able to pickup the endpoints anymore on the interface - which I use for "scanning".

Now I'm wondering if there's a way for Enunciate to detect that the interface actually IS a RestController without having that annotation set on it directly - but maybe looking at the implementation class which has it? Does the internal metamodel in Enunciate already has this kind of information?

**Comment by vghero** [comment]
I agree it's a bit "tricky" and surely does not make it less complex. Naive me thought just looking for the implementation class of the interface. But if we have multiple impls I would expect that all of them will be selected as a target for documentation generation. Similar to what happens with referenced DTOs in an interface - they are also picked up although maybe lying out of the provided package to scan. If this is a bigger thing todo it's maybe not worth the effort....

**Comment by stoicflame** [comment]
I can take a closer look, but let me summarize to make sure I understand what you'd like to see.

* Currently impl controller classes are getting picked up for documentation by Enunciate, but interfaces are only getting picked up if they're annotated with `@RestController`.
* You'd like to see Enunciate pick up interfaces as controllers if they have `@[Method]Mapping` annotations on their methods and if they have an impl class that is annotated with `@RestController`.

Is that accurate?

**Comment by vghero** [comment]
Yes, that is accurate :).
